include/ShorSimulator.hpp

```diff
@@ -77,6 +77,9 @@ class ShorSimulator : public Simulator {
     const bool emulate;
     const bool verbose;
     const bool approximate;
+    unsigned long long approximation_runs{0};
+    long double final_fidelity{1.0L};
+    double step_fidelity{0.9};
```

Comment:
Is it on purpose that these to have different types? I'd suppose `double` is enough for both of them.

Reply:
`double` should be enough and I'll change that for the next bigger approximation change, but for now I'd like to keep this for consistency.

Reply:
Sounds good 👍🏼